src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/services/NomisApiService.kt

```diff
@@ -19,27 +19,22 @@ class NomisApiService(@Qualifier("nomisApiWebClient") private val webClient: Web
     val log: Logger = LoggerFactory.getLogger(this::class.java)
   }
 
-  fun createVisit(request: CreateVisitDto) {
+  fun createVisit(request: CreateVisitDto): String? =
     webClient.post()
       .uri("/prisoners/${request.offenderNo}/visits")
       .bodyValue(request)
       .retrieve()
-      .bodyToMono(Unit::class.java)
-      .onErrorResume(WebClientResponseException.Conflict::class.java) {
-        log.warn("createVisit failed for offender no ${request.offenderNo} and vsip visit id ${request.vsipVisitId}  with message ${it.message}")
-        Mono.empty()
-      }
+      .bodyToMono(String::class.java)
```

Comment:
Does "create" just return a string? I would have thought it would return an object for the new visit, at least an object with a visit id in rather just a non-JSON string.

That is the pattern we are using for creating VSIP visits - tends to be more flexible to return an JSON object that a primitive String

Reply:
Correct - it does return an object, will fix